coverage/src/test/kotlin/org/rust/coverage/RsCoverageTest.kt

```diff
@@ -0,0 +1,222 @@
+            authors = []
+        """)
+
+        dir("src") {
+            rust("lib.rs", """
+                fn foo() {                      // Hits: 4
+                    println!("Hello, world!");  // Hits: 4
+                }                               // Hits: 4
+                
+                fn main() {                     // Hits: 1
+                    for _ in 1..5 {             // Hits: 5
+                        foo();                  // Hits: 4
+                    }
+                }                               // Hits: 1
+            """)
+        }
+    }
+
+    fun `test tests`() = doTest(runTests = true) {
+        toml("Cargo.toml", """
+            [package]
+            name = "hello"
+            version = "0.1.0"
+            authors = []
+        """)
+
+        dir("src") {
+            rust("lib.rs", """
+                fn foo() {                      // Hits: 2
+                    println!("Hello, world!");  // Hits: 2
+                }                               // Hits: 2
+                
+                #[test]
+                fn test1() {                    // Hits: 2
+                    foo();                      // Hits: 1
+                }                               // Hits: 2
+                
+                #[test]
+                fn test2() {                    // Hits: 2
+                    foo();                      // Hits: 1
+                }                               // Hits: 2
+            """)
+        }
+    }
+
+    private fun executeGrcov(configuration: RunConfiguration) {
+        val executor = DefaultRunExecutor.getRunExecutorInstance()
+        val environment = ExecutionEnvironmentBuilder.create(executor, configuration).build()
+        val state = environment.state as CargoRunStateBase
+        GrcovRunner().execute(environment, null, state)
+    }
+
+    private fun doTest(runTests: Boolean = false, builder: FileTreeBuilder.() -> Unit) {
+        buildProject(builder)
+
+        val expected = hashMapOf<String, List<Pair<Int, Int>>>()
+        VfsUtil.iterateChildrenRecursively(cargoProjectDirectory, null) { fileOrDir ->
+            if (!fileOrDir.isRustFile) return@iterateChildrenRecursively true
+            val psiFile = fileOrDir.toPsiFile(project) ?: return@iterateChildrenRecursively true
+            expected[fileOrDir.path] = hitsFrom(psiFile.text)
+            true
+        }
+
+        val configuration = createConfiguration()
+        if (runTests) {
+            configuration.command = "test"
+        }
+
+        executeGrcov(configuration)
+        runWithInvocationEventsDispatching("Failed to fetch coverage data") { lastProjectData != null }
+        val projectData = lastProjectData!!
+
+        val actual = hashMapOf<String, List<Pair<Int, Int>>>()
+        for (path in expected.keys) {
+            val data = projectData.getClassData(path)!!
+            actual[path] = hitsFrom(data)
+        }
+
+        assertEquals(expected, actual)
```

Comment:
Let's come up with more readable structure for comparison because currently if actual hits count differs from expected, you get something like
```
expected:<{/private/var/folders/hj/rsx5n85s56g3_mj0fgznct500000gp/T/unitTest__tests/unitTest7730444972665203395/src/lib.rs=[(1, 2), (2, 2), (3, 2), (6, 1), (7, 1), (8, 2), (11, 2), (12, 1), (13, 2)]}> but was:<{/private/var/folders/hj/rsx5n85s56g3_mj0fgznct500000gp/T/unitTest__tests/unitTest7730444972665203395/src/lib.rs=[(1, 2), (2, 2), (3, 2), (6, 2), (7, 1), (8, 2), (11, 2), (12, 1), (13, 2)]}>
```

And it's rather hard to understand what's wrong here